modules/mysql/mysql.go

```diff
@@ -0,0 +1,90 @@
+package mysql
+
+import (
+	"context"
+	"fmt"
+	"github.com/testcontainers/testcontainers-go"
+	"github.com/testcontainers/testcontainers-go/wait"
+	"strings"
+)
+
+const rootUser = "root"
+const defaultUser = "test"
+const defaultPassword = "test"
+const defaultDatabaseName = "test"
+
+// MySQLContainer represents the MySQL container type used in the module
+type MySQLContainer struct {
+	testcontainers.Container
+	username string
+	password string
+	database string
+}
+
+type PostgresContainerOption func(req *testcontainers.ContainerRequest)
+
+// StartContainer creates an instance of the MySQL container type
+func StartContainer(ctx context.Context, image string, opts ...PostgresContainerOption) (*MySQLContainer, error) {
+	req := testcontainers.ContainerRequest{
+		Image:        image,
+		ExposedPorts: []string{"3306/tcp", "33060/tcp"},
+		Env: map[string]string{
+			"MYSQL_USER":     defaultUser,
+			"MYSQL_PASSWORD": defaultPassword,
+			"MYSQL_DATABASE": defaultDatabaseName,
+		},
+		WaitingFor: wait.ForLog("port: 3306  MySQL Community Server"),
+	}
+
+	opts = append(opts, func(req *testcontainers.ContainerRequest) {
+		username := req.Env["MYSQL_USER"]
+		password := req.Env["MYSQL_PASSWORD"]
+		if strings.EqualFold(rootUser, username) {
+			delete(req.Env, "MYSQL_USER")
+		}
```

Comment:
This block is probably not needed, as setting the env var as `root` could be something harmless

Reply:
it is indeed 😅 otherwise, the container will not start. That's the reason why the env var should not be present at all